### PR TITLE
Enhance path build expression

### PR DESCRIPTION
**Pull request by yixinglu**
Now we can use path builder expression to build the path for following cases:
1. V
2. V, E
3. V, Path
4. Path, V, E
5. Path, Path
6. Path, V, Path

### CODE REVIEW

**Review by nevermore3 — APPROVED** [review]

File: src/common/function/FunctionManager.cpp [hunk 1]
```diff
@@ -1182,6 +1189,35 @@ FunctionManager::FunctionManager() {
             return result;
         };
     }
+    {
+        auto &attr = functions_["cyclePath"];
```
Comment (CPWstatic) [inline src/common/function/FunctionManager.cpp:1193]: According to the implementation, the function name should not be named `cyclePath`.  What it really want to do is to find if there exists same edges. So name it like `sameEdgeInPath` or something else properly.

**Review by CPWstatic — COMMENTED** [review]
LGTM.

File: src/common/expression/test/ExpressionTest.cpp [hunk 2]
```diff
@@ -2732,13 +2781,77 @@ TEST_F(ExpressionTest, PathBuild) {
...
+
+    {
+        auto expr0 = std::make_unique<PathBuildExpression>();
+        expr0->add(varPropExpr("path_src"));
+        auto expr = std::make_unique<PathBuildExpression>();
+        expr->add(std::move(expr0));
+        expr->add(varPropExpr("path_edge1"));
+
+        {
+            // Test: Path + Edge
```
[the remark this answers is not in the document]
Comment (yixinglu) [reply src/common/expression/test/ExpressionTest.cpp:2807]: in this case,  the expression will check whether the last vertex vid of path and source node of edge is same and the new path will contains a vertex which has no any properties after executing expression eval function.

File: src/common/function/FunctionManager.cpp (hunk 1 above)
Comment (yixinglu) [reply src/common/function/FunctionManager.cpp]: make sense. fixed it.

Comment (yixinglu) [reply src/common/function/FunctionManager.cpp]: named `hasSameEdgeInPath`

**Review by CPWstatic — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**